src/tunnel_controller.rs

```diff
@@ -89,3 +96,52 @@ impl Tunnel {
         }
     }
 }
+
+
+#[cfg(test)]
+describe! tunnel {
+
+    before_each {
+        use std::thread::sleep;
+        use std::time::Duration;
+        use stubs::tunnel::TunnelConfigStub;
+
+        let config = TunnelConfigStub::new(1, "a".to_owned());
+        let mut tunnel = super::Tunnel::new(config.clone());
+
+        tunnel.start().unwrap();
+        // XXX Sleep needed otherwise the process is not started in time
```

Comment:
Ah, start doesn't wait for the process to be spawned in `ManagedProcess`.  `start` kicks off a thread, which will eventually start the process.

We could move these stats (https://github.com/fxbox/foxbox/blob/e78390d2b46447805a1c316a110075af27826970/src/managed_process.rs#L66) to an `Arc<RwLock<_>>` on the ManagedProcess and have an API to inspect current backoff and process starts from the controlling thread.